pkg/ebpf/tracee.go

```diff
@@ -1572,6 +1572,11 @@ func (t *Tracee) triggerMemDump(event trace.Event) []error {
 
 	// TODO: consider to iterate over given policies when policies are changed
 	for p := range t.config.Policies.Map() {
+		// This might break in the future if PrintMemDump will become a dependency of another event.
+		_, isChosen := p.EventsToTrace[events.PrintMemDump]
```

Comment:
@geyslan correct me if I'm wrong, but I think this is not the best way to handle this issue.
Event states should be used instead.

Reply:
You're right, Yaniv. I'm attempting to change how EventStates are managed with #3848. For these peculiar events (PrintMemDump, HookedSyscall, ...) I've put comments for further efforts.

https://github.com/aquasecurity/tracee/pull/3848/files#diff-93a0a7535081b0454fec716306a8f60ab9de4de5d300d7799d8792eb0865e87cR218-R233

Reply:
That PR takes care of keeping the states per policies version:

![image](https://github.com/aquasecurity/tracee/assets/3372117/45d49797-535a-4216-9c95-3e2065a7879c)

In future efforts we'll have a policyManager (system wise = all policies versions in activity) aware of all events states.

The most relevant discussion for now is whether this type of event like PrintMemDump (initial) will only be necessary initially. If so, I would consider detaching their logic from Policies, because they fall into categories of option flags and not events to be managed by policies and so on. WDYT?

Reply:
Anyways, this PR is a fix of an issue we now have.
@yanivagman I see no reason not to fix it like this for now, and when we have the policies events states use it instead.